src/app/components/mod/mod.component.ts

```diff
@@ -0,0 +1,133 @@
+import { Component, Input } from '@angular/core';
+
+import { EditorialConventionLayoutData } from 'src/app/directives/editorial-convention-layout.directive';
+import { ChangeLayerData, Mod, Note } from 'src/app/models/evt-models';
+import { register } from 'src/app/services/component-register.service';
+import { EditionlevelSusceptible, Highlightable, ShowDeletionsSusceptible, TextFlowSusceptible } from '../components-mixins';
+import { distinctUntilChanged, map, scan, startWith, Subject } from 'rxjs';
+import { EVTStatusService } from 'src/app/services/evt-status.service';
+import { AppConfig, EditionLevelType } from 'src/app/app.config';
+
+export interface ModComponent extends EditionlevelSusceptible, Highlightable, TextFlowSusceptible, ShowDeletionsSusceptible { }
+
+@Component({
+  selector: 'evt-mod',
+  templateUrl: './mod.component.html',
+  styleUrls: ['./mod.component.scss'],
+})
+
+@register(Mod)
+export class ModComponent {
+
+  @Input() data: Mod;
+
+  @Input() editionLevel: EditionLevelType;
+
+  @Input() withDeletions: boolean;
+
+  public alwaysShow: boolean;
+
+  @Input() set alwaysShown(yes: boolean) {
+    this.alwaysShow = yes;
+  }
+  get alwaysShown() { return this.alwaysShow; }
+
+  public changeSeparatorVisible = AppConfig.evtSettings.edition.showSeparatorBetweenChanges;
+
+  public showLayerMarkers = AppConfig.evtSettings.edition.showChangeLayerMarkerInText;
+
+  public elementsToExcludeInTextFlow = [Note];
+
+  public orderedLayers: string[];
+
+  public selectedLayer: string;
+
+  public selectedLayer$ = this.evtStatusService.currentChanges$.pipe(
+    distinctUntilChanged(),
+    map(({ selectedLayer, layerOrder }) => {
+      this.selectedLayer = selectedLayer;
+      this.orderedLayers = layerOrder;
+      if (layerOrder.length > 0) {
+        this.selectedLayer = this.selectedLayer ?? layerOrder[layerOrder.length-1];
+      }
+
+      return selectedLayer;
+    } ),
+  );
+
+  public isHidden = this.layerHidden;
+
+  public changeLayerColor = this.getLayerColor;
+
+  toggleOpened$ = new Subject<boolean | void>();
+  opened$ = this.toggleOpened$.pipe(
+    scan((currentState: boolean, val: boolean | undefined) => val === undefined ? !currentState : val, false),
+    startWith(false),
+  );
+
+  get editorialConventionData(): EditorialConventionLayoutData {
```

Comment:
Change into a variable

Reply:
Seems dangerous at this time of the development, not taking this responsability for zero benefits